mockgcp/tools/patch-proto/main.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	sitter "github.com/smacker/go-tree-sitter"
```

Comment:
Just curious, why do you say the treesitter is an overkill? At the beginning I thought it means a lot of code, but it seems to be fine. Does this run take a long time to complete?